mako/codegen.py

```diff
@@ -1016,35 +1011,34 @@ class _Identifiers:
     """tracks the status of identifier names as template code is rendered."""
 
     def __init__(self, compiler, node=None, parent=None, nested=False):
-        if parent is not None:
+        if (
+            parent is not None
+            and isinstance(node, parsetree.NamespaceTag)
+            or parent is None
+        ):
             # if we are the branch created in write_namespaces(),
```

Comment:
**Michael Bourke** (bourke) wrote:

Same as 524-529 above. Looks cleaner, but is harder to understand.

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/mako/+/3319